### PR TITLE
Fixes #8214: 7844 breaks pdf build

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8214

### CODE REVIEW

**Comment by amousset** [comment]
The generic methods documentation:
* Breaks pdflatex (apparently because of the "$" chars), but I could not find any workaround for now
* Make the one page html quite longer, and looks bad with the current style

This PR limits it to the webhelp output.

File: 5_references/50_generic_methods.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+ifdef::webhelp[]
+
 === Generic methods
 
-include::../generic_methods.asciidoc[]
\ No newline at end of file
+include::../generic_methods.asciidoc[]
+
+endif::webhelp[]
```
Comment (jooooooon) [inline 5_references/50_generic_methods.txt]: Why can't we have the generic methods in the PDF too? It is build from asciidoc sources, what is wrong with this file?

I read the ticket, but there is no information in there. Please *always* explain the original problem in your tickets. Just having the solution is like only reading the second half of a book - it doesn't mean much to the reader.

**Comment by amousset** [comment]
PR updated

**Comment by jooooooon** [comment]
:+1:

**Pull request merged**